domain/watcher.go

```diff
@@ -75,8 +75,39 @@ func (f *WatcherFactory) NewNamespaceMapperWatcher(
 	), nil
 }
 
+// NewFilterWatcher returns a watcher that filters events based on the input
+// filter options. The watcher will only emit events that match the filter in
+// a single term.
+func (f *WatcherFactory) NewFilterWatcher(
+	filterOpts ...eventsource.FilterOption,
+) (watcher.NotifyWatcher, error) {
+	base, err := f.newBaseWatcher()
+	if err != nil {
+		return nil, errors.Annotate(err, "creating base watcher")
+	}
+
+	return eventsource.NewFilterWatcher(base, filterOpts...), nil
+}
+
+// NewFilterWatcher returns a watcher that filters events based on the input
+// filter options. The watcher will only emit events that match the filter in
+// a single term. The mapper is used to transform the events before they are
+// emitted, either by changing the event or by filtering them out.
+func (f *WatcherFactory) NewFilterMapperWatcher(
```

Comment:
These are the only ones that return `NotifyWatcher`. I'd just name them `NewNotifyWatcher` and `NewNotifyMapperWatcher`. Same with the base ones in `eventsource`.